src/model/PoolState.ts

```diff
@@ -0,0 +1,38 @@
+import { ConfigurableCorePool } from "../core/ConfigurableCorePool";
+import { PoolConfig } from "../entity/PoolConfig";
+import { Record } from "../entity/Record";
+import { Snapshot } from "../entity/Snapshot";
+import { DBManager } from "../manager/DBManager";
+
+export class PoolState {
+  id: string = "0";
+  baseSnapshot: Snapshot | null;
+  snapshot: Snapshot | null = null;
```

Comment:
I read that `undefined` is preferred to `null`.

See: https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#use-optional-parameters

Reply:
Could you please tell more about why undefined is preferred to null in this section?

Reply:
Forget it, I think I see your point.

Reply:
https://stackoverflow.com/questions/5076944/what-is-the-difference-between-null-and-undefined-in-javascript

Reply:
well I think null is better to use here for meaning part and undefined is acceptable for its practice.